.pre-commit-config.yaml

```diff
@@ -16,16 +16,20 @@ repos:
     -   id: check-json
     -   id: check-yaml
 -   repo: git://github.com/pre-commit/mirrors-pylint
-    sha: v2.3.1
+    rev: v2.3.1
     hooks:
     -   id: pylint
         args:
         - --rcfile=.pylintrc
         - --reports=no
+        exclude: ^tests/resources/.*_todo.*$
 -   repo: local
     hooks:
     -   id: py.test
         name: py.test
-        language: system
-        entry: sh -c py.test
-        files: ''
\ No newline at end of file
+        language: python
+        additional_dependencies: ['pytest', 'pytest-cov', 'coverage', 'fuzzywuzzy', 'python-Levenshtein']
```

Comment:
I think there may be a bit of a confusion here:
- this file describe the hooks to **apply** on this project, not the ones it **exposes**.
`.pre-commit-hooks.yaml` is the file containing the exposed hooks definitions.
- here those `additional_dependencies` are only specific to the `py.test` hook.
So no need for `fuzzywuzzy` & `python-Levenshtein`

Reply:
Yeah I realise the difference and I still think it is realy nice to have those dependencies here. 

In this case the `additional dependencies` are used if you want to run "git commit" without activating your own dev virtualenv (with dev requirements installed).

This is rather nice. We are runnning all tests in this pre-commit hook (py.test) - including fuzzy matchibg, we need to have fuzzywuzzy and python-Levenshtein installed in your environment. You simply check-out the project, make some changes and then run git-commit and that's it. No need to activate virtualenv with installed extra dev requirements. Try it out :)

This is the magic of new versions of pre-commit with 'additional_dependencies' - when you set language=python, then pre-commit will create a separate, new virtualenv for you under the hood and install those dependencies - so that all the tests works like magic.

Reply:
I know about how `pre-commit` work with virtualenvs under the hood.

I'm still a bit relunctant about his change, as it somehow hides dependency management logic in files that are not dedicated to this, but I see the benefit. And it **is** related to the tests executed by the `pytest` hook so OK, lets go with this.